diff --git a/sites/all/modules/contrib/robotstxt/robotstxt.module b/Users/grotter/Desktop/robotstxt.module
index 97f7e43..1cb4423 100644
--- a/sites/all/modules/contrib/robotstxt/robotstxt.module
+++ b/Users/grotter/Desktop/robotstxt.module
@@ -69,10 +69,6 @@ function robotstxt_robots() {
   $content = array_filter($content);
 
   drupal_add_http_header('Content-type', 'text/plain');
-
-  // grotter
-  die(implode("\n", $content));
-
-  // echo implode("\n", $content);
-  // drupal_page_footer();
+  echo implode("\n", $content);
+  drupal_page_footer();
 }
