diff --git a/googlenews.admin.inc b/googlenews.admin.inc
index 48451e5..40b1640 100644
--- a/googlenews.admin.inc
+++ b/googlenews.admin.inc
@@ -32,6 +32,32 @@ function googlenews_admin_settings() {
     '#default_value' => variable_get('googlenews_node_types', array_keys($node_types)),
     '#options' => $node_types,
   );
+
+  // @todo
+  // these checkboxes should be inserted with AJAX and only apply to
+  // associated content types
+  $vocabularies = taxonomy_get_vocabularies('explore_science_article');
+  $voc = array();
+
+  foreach ($vocabularies as $vocabulary) {
+    if ($vocabulary->machine_name == 'explore_science_categories') {
+      $tree = taxonomy_get_tree($vocabulary->vid);
+
+      foreach ($tree as $term) {
+        $voc[$term->tid] = t($term->name);
+      }
+    }
+  }
+
+  // asort($voc);
+
+  $form['googlenews_terms'] = array(
+    '#type' => 'checkboxes',
+    '#title' => t('Limit articles to those categorized as one of following:'),
+    '#default_value' => variable_get('googlenews_terms'),
+    '#options' => $voc,
+  );
+
   $form['array_filter'] = array(
     '#type' => 'value',
     '#value' => TRUE,
diff --git a/googlenews.install b/googlenews.install
index 52ad19c..91d3810 100644
--- a/googlenews.install
+++ b/googlenews.install
@@ -9,6 +9,7 @@
  * Implementation of googlenews_uninstall().
  */
 function googlenews_uninstall() {
+  variable_del('googlenews_terms');
   variable_del('googlenews_sitemap_url');
   variable_del('googlenews_node_types');
   variable_del('googlenews_cache_timeout');
diff --git a/googlenews.module b/googlenews.module
index 12e10a0..7aae2d3 100644
--- a/googlenews.module
+++ b/googlenews.module
@@ -103,6 +103,23 @@ function googlenews_list_nodes() {
 
   $query = db_select('node', 'n');
   $query->fields('n', array('nid'));
+
+  // filter by term
+  $tids = variable_get('googlenews_terms');
+
+  if (is_array($tids) && !empty($tids)) {
+    $or = db_or();
+
+    foreach ($tids as $tid) {
+      $or->condition('field_es_category_tid', $tid);
+    }
+
+    // @todo
+    // field should be dynamic
+    $query->join('field_data_field_es_category', 't', 't.entity_id=n.nid');
+    $query->condition($or);
+  }
+
   $query->condition('n.type', variable_get('googlenews_node_types', array_keys(node_type_get_names())));
   $query->condition('n.status', NODE_PUBLISHED);
   $query->condition('n.created', $time, '>=');
@@ -110,5 +127,6 @@ function googlenews_list_nodes() {
   $query->addTag('googlenews_sitemap');
   $query->orderBy('n.created', 'DESC');
   $query->range(0, 1000);
+
   return $query->execute()->fetchAll();
 }
