--- /Users/jause/projects/cas/sites/all/modules/contrib/manualcrop/manualcrop.module	2014-12-08 13:50:38.000000000 -0800
+++ manualcrop.module	2014-12-08 14:11:45.000000000 -0800
@@ -649,11 +649,12 @@
     // Loop trough the selections to save them. $data['styles'] can by an empty array,
     // but we have to loop trough the whole array to delete old selections.
     foreach ($form_state['manualcrop_data']['selections'] as $fid => $data) {
-      db_delete('manualcrop')
-        ->condition('fid', $fid)
-        ->execute();
-
       foreach ($data['styles'] as $style_name => $selection) {
+        // only delete crop styles for crop selections targeted in form object
+        db_delete('manualcrop')
+          ->condition('fid', $fid)
+          ->condition('style_name', $style_name)
+          ->execute();
         db_insert('manualcrop')
           ->fields(array_merge($selection, array(
             'fid' => $fid,
